### PR TITLE
Add a script to generate Swift API reference

### PR DESCRIPTION
**Pull request by danielnugraha**
### Description

The Swift API reference needs to be generated and converted into a static html page, ready to be published alongside the Python API reference.

### Explanation

Add a script build-swift-api-ref.sh to generate API reference .doccarchive and convert it into static html page using swift docc plugin.

### Checklist

- [x] Implement proposed change
- [x] Check if everything is alright
- [ ] Integrate the script into the CI
- [ ] Make CI checks pass

### CODE REVIEW

**Comment by tanertopal** [comment]
@danielnugraha I encountered this issue:

```bash
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
```

Which was solved as described here: https://github.com/nodejs/node-gyp/issues/569

Did you also encounter that issue? Should we document it in a comment?

File: dev/build-swift-api-ref.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# making sure to start a clean build
+find ~/Library/Developer/Xcode/DerivedData -name "flwr.doccarchive" -exec rm -Rf {} \; || true
+
+# change directory to the swift sdk source code folder
+cd src/swift/flwr
+# generate api reference for swift sdk
+arch -x86_64 xcodebuild docbuild -scheme flwr -destination 'platform=iOS Simulator,name=iPhone 14 Pro Max,OS=16.4'
+
+# go back to root folder
+cd ../../../
```
Comment (tanertopal) [inline dev/build-swift-api-ref.sh:32]:
```suggestion
# Generate API reference for the Swift SDK by running `xcodebuild docbuild` in src directory
cd src/swift/flwr && \
arch -x86_64 xcodebuild docbuild -scheme flwr -destination 'platform=iOS Simulator,name=iPhone 14 Pro Max,OS=16.4'

```

File: dev/build-swift-api-ref.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+set -e
+cd "$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/../
+
+# create a directory to save generated api reference
```
Comment (tanertopal) [inline dev/build-swift-api-ref.sh:21]:
```suggestion
# Create a directory to save generated API reference
```

File: dev/build-swift-api-ref.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+mkdir -p SwiftDoc
+# making sure to start a clean build
```
Comment (tanertopal) [inline dev/build-swift-api-ref.sh:23]:
```suggestion
# Make sure to start a clean build by deleting all potentially old artefacts
```

File: dev/build-swift-api-ref.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+find ~/Library/Developer/Xcode/DerivedData -name "flwr.doccarchive" -exec rm -Rf {} \; || true
+
+# change directory to the swift sdk source code folder
+cd src/swift/flwr
+# generate api reference for swift sdk
+arch -x86_64 xcodebuild docbuild -scheme flwr -destination 'platform=iOS Simulator,name=iPhone 14 Pro Max,OS=16.4'
+
+# go back to root folder
+cd ../../../
+# find the generated doccarchive file in Xcode derived data folder and copy it to the SwiftDoc directory
```
Comment (tanertopal) [inline dev/build-swift-api-ref.sh:33]:
```suggestion
# Find the generated doccarchive file in XCode's derived data folder and copy it to the SwiftDoc directory
```

**Comment by tanertopal** [comment]
@danielnugraha Took me a while to understand that the path we have to open is `http://localhost:3000/documentation/flwr` when starting a server in `SwiftDoc/html`. Any chance you can write some guide for this?

**Comment by danielnugraha** [comment]
> @danielnugraha I encountered this issue:
> 
> ```shell
> xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance
> ```
> 
> Which was solved as described here: [nodejs/node-gyp#569](https://github.com/nodejs/node-gyp/issues/569)
> 
> Did you also encounter that issue? Should we document it in a comment?

It only occured if you don't have any Xcode installed in your Mac - I will add comment in the script.

**Comment by danielnugraha** [comment]
> @danielnugraha Took me a while to understand that the path we have to open is `http://localhost:3000/documentation/flwr` when starting a server in `SwiftDoc/html`. Any chance you can write some guide for this?

I just add some comment in the script? If we publish it the path will be hidden from user perspective. There is also an option to change the path in the documentation generation.

**Review by tanertopal — APPROVED** [review]

**Pull request merged**